xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/internal/macro/AbstractShowHideIfMacro.java

```diff
@@ -128,4 +138,36 @@ protected boolean doesMatch(ShowHideIfMacroParameters parameters)
         return (parameters.getMatchUsing() == ShowHideIfMacroParameters.Matcher.ANY && matchAnyRes)
             || (parameters.getMatchUsing() == ShowHideIfMacroParameters.Matcher.ALL && matchAllRes);
     }
+
+    /**
+     * We need to check if all passed parameter are supported by XWiki because this macro could be imported from
+     * confluence and on confluence a lot more parameter are supported. In case of the parameter is not supported it
+     * could confuse the user thy this macro don't work as expected.
+     *
+     * @param context
+     * @return
+     */
+    protected Optional<Block> maybeGetUnsupportedParameterErrorBlock(MacroTransformationContext context)
+    {
+        List<String> parametersWhiteList = List.of("atlassian-macro-output-type");
+        java.util.Map<String, String> allParameters = context.getCurrentMacroBlock().getParameters();
+        Set<String> parameterNames = allParameters.keySet();
+        BeanDescriptor beanDescriptor = beanManager.getBeanDescriptor(ShowHideIfMacroParameters.class);
+        List<String> beanPropertiesIds =
+            beanDescriptor.getProperties().stream().map(PropertyDescriptor::getId).collect(Collectors.toList());
+        List<String> unsupportedParameters = new LinkedList<>();
+        for (String parameterName : parameterNames) {
+            if (!beanPropertiesIds.contains(parameterName) && !parametersWhiteList.contains(parameterName)) {
+                unsupportedParameters.add(parameterName);
+            }
+        }
+        if (!unsupportedParameters.isEmpty()) {
+            return Optional.of(
+                new MacroBlock("error", Collections.emptyMap(),
+                    // TODO escape content but seem that XWikiSyntaxEscaper
+                    //  is only available since XWiki-platform 14.10.6
+                    "Unsupported parameter for macro: " + String.join(", ", unsupportedParameters), false));
```

Comment:
@trrenty  do you have any idea how to manage escaping for a macro content when we are on XWiki 14.10.0 ?